esp32/modsocket.c

```diff
@@ -477,8 +478,20 @@ STATIC mp_obj_t esp_socket_getaddrinfo(const mp_obj_t host, const mp_obj_t port)
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_2(esp_socket_getaddrinfo_obj, esp_socket_getaddrinfo);
 
+STATIC mp_obj_t esp_socket_initialize() {
+    static int initialized = 0;
+    if (!initialized) {
+        ESP_LOGI("modsocket", "Initializing");
+        tcpip_adapter_init();
```

Comment:
Yep, but it `tcpip_adapter_init` checks if it has been called already and if so does nothing.  As an alternative, we could always init the stack on startup, but it seemed like it'd be better to wait until it was actually needed.